### PR TITLE
Do not fail empty result files when tags are specified.

### PR DESCRIPTION
**Pull request by double16**
Tags may be specified to narrow down test execution to fix a particular
problem. It is very likely, and intentional, that all features in a file
will be excluded and produce an empty result file. This should not be
considered an error condition.

### CODE REVIEW

**Comment by double16** [comment]
Any insight on how to determine the difference between a failed feature and ignored besides an empty result file?

File: src/main/groovy/com/commercehub/gradle/cucumber/CucumberRunner.groovy [hunk 1]
```diff
@@ -85,6 +88,10 @@ class CucumberRunner {
         return !testResultCounter.hadFailures()
     }
 
+    private boolean isResultFileEmpty(File resultsFile) {
+        resultsFile.size() < 8 && resultsFile.text.replaceAll(/\s+/, '') == '[]'
```
Comment (JayStGelais) [inline src/main/groovy/com/commercehub/gradle/cucumber/CucumberRunner.groovy]: '8' looks like a magic number here. Perhaps we can get a constant with a name that describes its meaning here.

File: src/main/groovy/com/commercehub/gradle/cucumber/CucumberRunner.groovy [hunk 2]
```diff
@@ -22,6 +22,13 @@ class CucumberRunner {
     private static final String TILDE = '~'
     private static final String TAGS = '--tags '
 
+    /**
+     * Result files can be quite large when including embedded images, etc. When checking if a result file is empty,
+     * we don't want to process large files. Any file with a size of this or larger won't be considered when checking
+     * for an empty result.
```
Comment (JayStGelais) [inline src/main/groovy/com/commercehub/gradle/cucumber/CucumberRunner.groovy]: Can we get a little more detail on this. It's still not clear to me why 8 was chosen as the value. Can a file have a size of 7 and still be empty?

Is this 8 bytes, 8kb, 8mb, etc...?

Comment (double16) [reply src/main/groovy/com/commercehub/gradle/cucumber/CucumberRunner.groovy]: The file size comparison is an inequality, so yes, a file with size 7 will still be checked as a valid but empty result. I added reasoning to the comment. I bumped up from 8 to 16 considering newlines on Windows being two bytes.

**Comment by JayStGelais** [comment]
@double16 Thanks for improved documentation on this change. This has been merged. I'll try to get a new version of the plugin deployed today.

**Pull request merged**